File: scaffold/src/main/java/top/xuqingquan/web/publics/JsAccessEntraceImpl.java
package top.xuqingquan.web.publics;

import android.os.Handler;
import android.os.Looper;

public class JsAccessEntraceImpl extends BaseJsAccessEntrace {

    private Handler mHandler = new Handler(Looper.getMainLooper());

    public static JsAccessEntraceImpl getInstance(android.webkit.WebView webView) {
        return new JsAccessEntraceImpl(webView);
    }

    public static JsAccessEntraceImpl getInstance(com.tencent.smtt.sdk.WebView webView) {
        return new JsAccessEntraceImpl(webView);
    }

    private JsAccessEntraceImpl(android.webkit.WebView webView) {
        super(webView);
    }

    private JsAccessEntraceImpl(com.tencent.smtt.sdk.WebView webView) {
        super(webView);
    }

    private void safeCallJs(final String s, final android.webkit.ValueCallback valueCallback) {
        mHandler.post(() -> callJs(s, valueCallback));
    }

    private void safeCallJs(final String s, final com.tencent.smtt.sdk.ValueCallback valueCallback) {
        mHandler.post(() -> callJs(s, valueCallback));
    }

    @Override
    public void callJs(String params, final android.webkit.ValueCallback<String> callback) {
        if (Thread.currentThread() != Looper.getMainLooper().getThread()) {
            safeCallJs(params, callback);
            return;
        }
        super.callJs(params, callback);
    }

    @Override
    public void callJs(String params, final com.tencent.smtt.sdk.ValueCallback<String> callback) {
        if (Thread.currentThread() != Looper.getMainLooper().getThread()) {
            safeCallJs(params, callback);
            return;
        }
        super.callJs(params, callback);
    }


}

File: scaffold/src/main/java/top/xuqingquan/web/publics/JsAccessEntraceImpl.kt
package top.xuqingquan.web.publics

import android.os.Handler
import android.os.Looper

class JsAccessEntraceImpl : BaseJsAccessEntrace {

    private val mHandler = Handler(Looper.getMainLooper())

    private constructor(webView: android.webkit.WebView?) : super(webView)

    private constructor(webView: com.tencent.smtt.sdk.WebView?) : super(webView)

    private fun safeCallJs(s: String?, valueCallback: android.webkit.ValueCallback<String>?) {
        mHandler.post { callJs(s, valueCallback) }
    }

    private fun safeCallJs(s: String?, valueCallback: com.tencent.smtt.sdk.ValueCallback<String>?) {
        mHandler.post { callJs(s, valueCallback) }
    }

    override fun callJs(js: String?, callback: android.webkit.ValueCallback<String>?) {
        if (Thread.currentThread() !== Looper.getMainLooper().thread) {
            safeCallJs(js, callback)
            return
        }
        super.callJs(js, callback)
    }

    override fun callJs(js: String?, callback: com.tencent.smtt.sdk.ValueCallback<String>?) {
        if (Thread.currentThread() !== Looper.getMainLooper().thread) {
            safeCallJs(js, callback)
            return
        }
        super.callJs(js, callback)
    }

    companion object {

        @JvmStatic
        fun getInstance(webView: android.webkit.WebView?): JsAccessEntraceImpl {
            return JsAccessEntraceImpl(webView)
        }

        @JvmStatic
        fun getInstance(webView: com.tencent.smtt.sdk.WebView?): JsAccessEntraceImpl {
            return JsAccessEntraceImpl(webView)
        }
    }


}
